components/Graphs/MostUsedLanguagesGraph/most-used-languages-graph.tsx

```diff
@@ -0,0 +1,134 @@
+import { BsFillCircleFill } from "react-icons/bs";
+import Skeleton from "react-loading-skeleton";
+import { useEffect, useRef, useState } from "react";
+import Card from "components/atoms/Card/card";
+import Text from "components/atoms/Typography/text";
+import { ContributorType, ContributorTypeFilter } from "../shared/contributor-type-filter";
+
+export interface MostUsedLanguagesGraphProps {
+  data: {
+    mainLanguage: string;
+    data: {
+      name: string;
+      value: number;
+    }[];
+  };
+  setContributorType: (type: ContributorType) => void;
+  contributorType: ContributorType;
+  isLoading?: boolean;
+}
+
+export const MostUsedLanguagesGraph = ({
+  data,
+  setContributorType,
+  contributorType,
+  isLoading = false,
+}: MostUsedLanguagesGraphProps) => {
+  const colors = [
+    "hsl(53, 91%, 59%)",
+    "hsl(204, 100%, 40%)",
+    "hsl(14, 98%, 49%)",
+    "hsl(267, 36%, 37%)",
+    "hsl(17, 100%, 50%)",
+  ];
+  const percentage = 10;
+  const { data: languages = [], mainLanguage } = data;
+  const lastItem = languages.length > 0 ? languages.length - 1 : 0;
+  const sortedLanguages = languages.sort((a, b) => b.value - a.value);
+  const languagesRef = useRef<HTMLUListElement>(null);
+  const [language, setLanguage] = useState<string | null>();
+
+  useEffect(() => {
+    if (language) {
+      const languageElement = languagesRef.current?.querySelector(`[data-language="${language}"]`);
+      if (languageElement) {
+        languageElement.classList.add("font-semibold");
+      }
+    }
+  }, [language]);
+
+  return (
+    <Card className="p-5">
+      <div className="flex flex-col gap-6">
+        <div>
+          <h2 className="pb-1 font-medium text-lg tracking-tight">Most used languages</h2>
+          <Text>
+            {mainLanguage} contributions have been growing on average {percentage}% MoM
+          </Text>
+        </div>
+        <div className="w-max">
+          <ContributorTypeFilter setContributorType={setContributorType} contributorType={contributorType} />
+        </div>
+
+        <div className="flex h-3 place-content-center">
+          {isLoading ? (
+            <div className="loading rounded-lg w-max" style={{ width: "100%" }}>
+              <span className="sr-only">loading most used languages graph</span>
+            </div>
+          ) : (
+            <>
+              {sortedLanguages.length > 0 ? (
+                sortedLanguages.map((item, index) => {
+                  return (
+                    <button
+                      aria-label={`${item.name} is ${item.value}% of the most used languages for contributors in your list`}
+                      key={item.name}
+                      data-language={item.name}
```

Comment:
More on [data attributes](https://developer.mozilla.org/en-US/docs/Learn/HTML/Howto/Use_data_attributes) in my latest blog post, [HTML Data Attributes: One of the Original State Management Libraries](https://dev.to/nickytonline/html-data-attributes-one-of-the-original-state-management-libraries-8bf)